app/controllers/user/orders_controller.rb

```diff
@@ -16,7 +16,7 @@ def create
         order.order_items.create({
           item: item,
           quantity: cart.count_of(item.id),
-          price: item.price
+          price: item.find_price(@cart)
```

Comment:
I created the find price method for items to find the discounted price if a discount applies.